### PR TITLE
Sample perf events

### PR DESCRIPTION
**Pull request by andre4i**
## Description

Related to ADO:5682

This change adds the capability to sample measure performance events per (event name + event category) and enables this for the noisy event.

### CODE REVIEW

**Comment by andre4i** [comment]
@alexvy86 I have split this from https://github.com/microsoft/FluidFramework/pull/17518, as I noticed some issues with the stress test feature gates, so that PR will span too many domains for me to be comfortable with.

File: packages/utils/telemetry-utils/src/logger.ts [hunk 1]
```diff
@@ -566,22 +566,50 @@ export interface IPerformanceEventMarkers {
  * Helper class to log performance events
  */
 export class PerformanceEvent {
+	/**
+	 * Creates an instance of {@link PerformanceEvent} and start measurements
```
Comment (alexvy86) [inline packages/utils/telemetry-utils/src/logger.ts:570]:
```suggestion
	 * Creates an instance of {@link PerformanceEvent} and starts measurements
```

File: packages/utils/telemetry-utils/src/logger.ts [hunk 2]
```diff
@@ -566,22 +566,50 @@ export interface IPerformanceEventMarkers {
...
+	/**
+	 * Measure a synchronous task
+	 * @param logger - the logger to be used for publishing events
+	 * @param event - the logging event details which will be published with the performance measurements
+	 * @param callback - the task to be executed and measured
+	 * @param markers - See {@link IPerformanceEventMarkers}
+	 * @param sampleThreshold - events with the same name and category will be sent to the logger
+	 * only when we hit this many executions of the task. If unspecified, all events will be sent.
+	 * @returns The results of the executed task
+	 */
```
Comment (alexvy86) [inline packages/utils/telemetry-utils/src/logger.ts:598]: Might need word-wrapping adjustments, and should also go in `timedExecAsync`.
```suggestion
	 * @returns The results of the executed task
	 *
	 * @remarks Note that if the "same" event (category + eventName) would be emitted by different
	 * tasks (`callback`), `sampleThreshold` is still applied only based on the event's category + eventName, 
	 * so executing either of the tasks will increase the internal counter and they
	 * effectively "share" the sampling rate for the event.
	 */
```

File: packages/utils/telemetry-utils/src/logger.ts [hunk 3]
```diff
@@ -566,22 +566,50 @@ export interface IPerformanceEventMarkers {
...
  */
 export class PerformanceEvent {
+	/**
+	 * Creates an instance of {@link PerformanceEvent} and start measurements
+	 * @param logger - the logger to be used for publishing events
+	 * @param event - the logging event details which will be published with the performance measurements
+	 * @param markers - See {@link IPerformanceEventMarkers}
+	 * @param recordHeapSize - whether or not to also record memory performance
+	 * @param emitLogs - should this instance emit logs. If set to false, logs will not be emitted to the logger,
+	 * but measurements will still be performed
```
Comment (alexvy86) [inline packages/utils/telemetry-utils/src/logger.ts:576]:
```suggestion
	 * but measurements will still be performed and any specified markers will be generated.
```

**Review by alexvy86 — APPROVED** [review]
A few extra docs suggestions. Functionally lgtm.

**Pull request merged**